src/cpp/rtps/participant/RTPSParticipantImpl.cpp

```diff
@@ -238,6 +239,21 @@ RTPSParticipantImpl::RTPSParticipantImpl(
     // User defined transports
     for (const auto& transportDescriptor : m_att.userTransports)
     {
+        auto socket_descriptor =
+                std::dynamic_pointer_cast<fastdds::rtps::SocketTransportDescriptor>(transportDescriptor);
+        if (socket_descriptor != nullptr)
+        {
+            if (!fastdds::rtps::NetmaskFilterUtils::validate_and_transform(socket_descriptor->netmask_filter,
```

Comment:
Here should probably avoid modifying user-provided descriptor.

Reply:
Added a locking interface to `TransportDescriptorInterface` so it is possibly to safely modify for registration, and restoring original values after registration.